### PR TITLE
#1720 hook up patient edit modal

### LINKED ISSUE #1720: Hook PatientEdit modal up to patients table
## Is your feature request related to a problem? Please describe.

`PatientEdit` modal added in the dispensary branch isn't accessible.

## Describe the solution you'd like

When clicking on a row on the `Patients` table, open the `PatientEdit` modal

## Implementation

N/A

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1720 

## Change summary

- Adds callbacks to open a patient modal in the dispensing page.

Couple offroadish: Added a little safer call in `sortDataBy` and also renamed `email` and `phone` to be consistent with the database/schema naming.

editingpatient

## Testing

N/A

### Related areas to think about

Wondering if this wouldv'e been better as a more generic form type component. I think it would be, probably worth changing..!

### CODE REVIEW

**Comment by joshxg** [comment]
I am working on #1724 now, which changes this relatively significantly. I also branched off this rather than create something completely new. I am not sure if this is the best decision

File: src/pages/DispensingPage.js [hunk 1]
```diff
@@ -86,24 +92,29 @@ const Dispensing = ({
...
+    <>
+      <DataTablePageView>
+        <View style={pageTopSectionContainer}>
+          <ToggleBar toggles={toggles} />
+          <SearchBar
+            onChangeText={onFilterData}
+            value={searchTerm}
+            viewStyle={localStyles.searchBar}
+          />
+          <PageButton text="New Patient" onPress={createPatient} />
```
Comment (mark-prins) [inline src/pages/DispensingPage.js]: this is in the 'translate me' issue?

File: src/reducers/PatientReducer.js [hunk 2]
```diff
@@ -42,10 +49,50 @@ export const PatientReducer = (state = patientInitialState(), action) => {
     case PATIENT_ACTIONS.FIELD_UPDATE: {
       const { payload } = action;
       const { field, value } = payload;
+      const { hasValue } = state;
+
+      const isRequired = field in hasValue;
+
+      if (isRequired) {
+        return { ...state, hasValue: { ...hasValue, [field]: !!value }, [field]: value };
```
Comment (mark-prins) [inline src/reducers/PatientReducer.js]: a very concise one-liner! that's beautiful

File: src/widgets/modals/PatientEdit.js [hunk 3]
```diff
@@ -97,8 +99,8 @@ export const PatientEditComponent = ({
               onChangeValidState={setDateOfBirthValidity}
               onSubmitEditing={setDateOfBirth}
             />
-            <ValidationTextInput value={email} label="Email" onSubmitEditing={setEmail} />
-            <ValidationTextInput value={phone} label="Phone" onSubmitEditing={setPhone} />
+            <ValidationTextInput value={emailAddress} label="Email" onSubmitEditing={setEmail} />
```
Comment (mark-prins) [inline src/widgets/modals/PatientEdit.js]: localisation?

**Pull request merged**